narwhals/_pandas_like/series.py

```diff
@@ -546,6 +546,11 @@ def second(self) -> PandasSeries:
             self._series._series.dt.second,
         )
 
+    def nanosecond(self) -> PandasSeries:
+        return self._series._from_series(
+            ((self._series._series.dt.microsecond * 1e3).astype(int)),
```

Comment:
we don't currently have a good way of constructing datetimes with nanoseconds, but here you'll also need the nanosecond component:
```
self._series._series.dt.microsecond * 1_000 + self._series._series.dt.nanosecond
```